opentelemetry-sdk/src/opentelemetry/sdk/resources/__init__.py

```diff
@@ -55,3 +60,48 @@ def __hash__(self):
 
 
 _EMPTY_RESOURCE = Resource({})
+
+
+class ResourceDetector:
+    def __init__(self, raise_on_error=False):
+        self.raise_on_error = raise_on_error
+
+    # pylint: disable=no-self-use
+    def detect(self) -> "Resource":
+        return _EMPTY_RESOURCE
+
+
+class OTELResourceDetector:
+    # pylint: disable=no-self-use
+    def detect(self) -> "Resource":
+        env_resources_items = os.environ.get("OTEL_RESOURCE")
+        env_resource_map = {}
+        if env_resources_items:
+            env_resource_map = {
+                key.strip(): value.strip()
+                for key, value in (
+                    item.split("=") for item in env_resources_items.split(",")
+                )
+            }
+        return Resource(env_resource_map)
+
+
+def get_aggregated_resources(
```

Comment:
It would be great if we could see some examples of this usage with custom `ResourceDetectors`

Reply:
EDIT: Or some documentation on what is needed if someone were to write their own ResourceDetector.

Reply:
I was thinking about adding a section on resources but wasn't too sure where to put it. https://opentelemetry-python.readthedocs.io/en/stable/getting-started.html. I'm thinking in examples?

Reply:
Hmm, we should really have a basic_tracer folder in docs/examples. Maybe you can create one and have `resource.py` as an example there?